subsys/nrf_security/Kconfig.legacy

```diff
@@ -76,81 +76,81 @@ config MBEDTLS_AES_DECRYPT_ALT
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_CIPHER_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_CMAC_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CMAC_CC3XX
 	default y
 
 config MBEDTLS_CCM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CCM_CC3XX
 	default y
 
 config MBEDTLS_GCM_ALT
 	bool
 	depends on CC312_BACKEND || \
-		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE)
+		   (PSA_CRYPTO_DRIVER_ALG_GCM_CC3XX && CRYPTOCELL_CC312_USABLE)
 	default y
 
 config MBEDTLS_CHACHA20_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 
 config MBEDTLS_POLY1305_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 
 config MBEDTLS_CHACHAPOLY_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX
 	default y
 	depends on MBEDTLS_CHACHA20_ALT
 
 config MBEDTLS_DHM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_RSA_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_ECP_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_ECDH_GEN_PUBLIC_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_ECDH_COMPUTE_SHARED_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_ECDSA_GENKEY_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_CC3XX
 	default y
 
 config MBEDTLS_ECDSA_SIGN_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_OBERON || \
+		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_HAS_ECC_SUPPORT_CC3XX
```

Comment:
Isn't that better fitted PSA_CRYPTO_DRIVER_ALG_ECDSA_CC3XX ? (Maybe also we can also use PSA_CRYPTO_DRIVER_ALG_DETERMINISTIC_ECDSA_CC3XX additionally)

Because the ECC_SUPPORT can also just mean ECDH